src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs

```diff
@@ -1375,6 +1375,13 @@ public virtual Expression BindAnyNode(AnyNode anyNode)
                 body = ApplyNullPropagationForFilterBody(body);
                 body = Expression.Lambda(body, anyIt);
             }
+            else if (anyNode.Body != null && anyNode.Body.Kind == QueryNodeKind.Constant
+                && (bool)(anyNode.Body as ConstantNode).Value == false)
```

Comment:
>(bool) [](start = 19, length = 6)

is this cast safe?

Reply:
Yes. We are checking anyNode.Body.Kind == QueryNodeKind.Constant first and casting only after that